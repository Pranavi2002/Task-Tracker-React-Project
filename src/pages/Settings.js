import React from "react";
import { Box, Card, CardContent, Typography, FormControlLabel, Switch } from "@mui/material";
import useTheme from "../hooks/useTheme";

export default function Settings() {
  const { mode, toggleTheme } = useTheme();

  return (
    <Box sx={{ p: 3, minHeight: "70vh", display: "flex", justifyContent: "center", alignItems: "center" }}>
      <Card sx={{ maxWidth: 400, width: "100%", bgcolor: "background.paper", color: "text.primary" }}>
        <CardContent>
          <Typography variant="h5" gutterBottom>
            Settings
          </Typography>
          <FormControlLabel
            control={<Switch checked={mode === "dark"} onChange={toggleTheme} />}
            label={mode === "light" ? "Light Mode" : "Dark Mode"}
            />
        </CardContent>
      </Card>
    </Box>
  );
}