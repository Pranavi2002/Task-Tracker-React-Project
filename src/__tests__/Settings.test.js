// src/__tests__/Settings.test.js
import React from "react";
import { render } from "@testing-library/react";
import Settings from "../pages/Settings";

// Mock useTheme hook
jest.mock("../hooks/useTheme", () => () => ({
  theme: "light",
  toggleTheme: jest.fn(),
}));

describe("Settings Snapshot", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<Settings />);
    expect(asFragment()).toMatchSnapshot(); // Capture DOM for theme toggle page
  });
});