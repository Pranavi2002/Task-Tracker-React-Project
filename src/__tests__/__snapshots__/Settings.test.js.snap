// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Snapshot renders correctly 1`] = `
<DocumentFragment>
  <div
    style="padding: 30px; min-height: 70vh; background-color: rgb(249, 249, 249); color: rgb(34, 34, 34); transition: 0.3s; text-align: center; display: flex; flex-direction: column; justify-content: center; align-items: center;"
  >
    <div
      style="background-color: rgb(255, 255, 255); padding: 20px 40px; border-radius: 12px; box-shadow: 0px 4px 12px rgba(0,0,0,0.2); max-width: 400px; width: 100%;"
    >
      <h1>
        Settings
      </h1>
      <div
        style="display: flex; justify-content: space-between; align-items: center; margin-top: 20px;"
      >
        <span>
          Light Mode
        </span>
        <div
          style="position: relative; display: inline-block; width: 60px; height: 34px;"
        >
          <div
            style="position: absolute; cursor: pointer; top: 0px; left: 0px; right: 0px; bottom: 0px; background-color: rgb(204, 204, 204); transition: .4s; border-radius: 34px;"
          />
          <div
            style="position: absolute; height: 26px; width: 26px; left: 4px; bottom: 4px; background-color: white; border-radius: 50%; transition: .4s;"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
